fuse/nodefs/fsops.go

```diff
@@ -70,6 +70,8 @@ func (c *FileSystemConnector) lookupMountUpdate(out *fuse.Attr, mount *fileSyste
 
 // internalLookup - execute a lookup without affecting NodeId reference counts
 func (c *FileSystemConnector) internalLookup(out *fuse.Attr, parent *Inode, name string, header *fuse.InHeader) (node *Inode, code fuse.Status) {
+	c.forgetMu.RLock()
```

Comment:
this is really problematic. You're holding a global FS lock while getattr executes. GetAttr is a user specified function, so it often does expensive or blocking actions (eg. network lookups.), so this lock would effectively kill performance. The fact that fsstress passes with this lock also doesn't say much, because the lock may mask some other race condition.